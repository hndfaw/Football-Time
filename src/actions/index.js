export const leaguesAction = leaguesData => ({
  type: 'SET_LEAGUES',
  leaguesData
})

export const setTodaysMatches = todayMatchesDate => ({
  type: 'SET_TODAYS_MATCHES',
  todayMatchesDate
})

export const setOneLeagueMatches = (leagueId, matches) => ({
  type: 'SET_ONE_LEAGUE_MATCHES',
  leagueId,
  matches
})