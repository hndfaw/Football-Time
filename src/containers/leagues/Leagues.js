import React, { Component } from 'react';
import { connect } from 'react-redux';
import './leagues.css';
import MatchContainer from '../matchesContainer/MatchesContainer';
import { fetchTodaysMatches } from '../../apiCalls';
import { setTodaysMatches } from '../../actions';



class Leagues extends Component {

  changeTodaysDate = e => {
    fetchTodaysMatches(e.target.value).then(data => this.props.handleTodaysMatches(data.api.fixtures))
  }


  render() {
    const {todaysMatches, leaguesData} = this.props;
    const filteredLeagus = [];
    todaysMatches.forEach(match => {
        leaguesData.forEach(league => {
          if (league.league_id === match.league_id && !filteredLeagus.includes(league)) {filteredLeagus.push(league) }
        })
    })
    
    const league = filteredLeagus.map(league => {

      return (
        <section key={league.league_id} id={league.league_id}>
        <div className="league-container">
          <div className="league-logo-container">
            <img src={league.logo} alt="league logo" className="league-logo" /> 
          </div>
          <h3 className="league-name">{league.name}</h3>
        </div>
        <MatchContainer id={league.league_id}/>
        </section>
      )
    })
    return (
      <main className="main">
        <div>
              <input type="date" onChange={this.changeTodaysDate}/>
        </div>
        {league} 
        leaguess....
      </main>
    )
  }
}

const mapStateToProps = state => ({
  leaguesData: state.leaguesData,
  todaysMatches: state.todaysMatches
})

const mapDispatchToProps = dispatch => ({
  handleTodaysMatches: data => dispatch(setTodaysMatches(data)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Leagues)