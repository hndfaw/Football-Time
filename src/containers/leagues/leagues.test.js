import React from 'react';
import {Leagues, mapDispatchToProps} from './Leagues';
import { shallow } from 'enzyme';
import { setSelectedLeague } from '../../actions';

describe('Leagues', () => {

  let wrapper;
  beforeEach(() => {

    const leaguesData = [
      {
        league_id: 524,
        logo: "https://www.api-football.com/public/leagues/2.png",
        name: "Premier League"
      }
    ]

    const mockMatch = [
      {
        awayTeamLogo: "https://www.api-football.com/public/teams/71.png",
        awayTeamName: "Norwich",
        elapsed: 0,
        event_date: "2019-08-09",
        fixture_id: 157015,
        goalsAwayTeam: null,
        goalsHomeTeam: null,
        homeTeamLogo: "https://www.api-football.com/public/teams/40.png",
        homeTeamName: "Liverpool",
        league_id: 524,
        status: "Not Started",
        statusShort: "NS"
      }
    ]
    wrapper = shallow(<Leagues dataLoading={mockMatch} selectedLeague="league524" league524={mockMatch} matches={mockMatch} id={{league_id: 524}} />);
  })

  it('should match the snopshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
 

it('should dispatch with a setSelectedLeague action when handleSelectedLeague is called', () => {
  const mockDispatch = jest.fn();
  const mockMatch = [{match: 'match'}];
  const mockAction = setSelectedLeague(mockMatch);
  const mappedProps = mapDispatchToProps(mockDispatch);
  mappedProps.handleSelectedLeague(mockMatch);
  expect(mockDispatch).toHaveBeenCalledWith(mockAction);
})

})