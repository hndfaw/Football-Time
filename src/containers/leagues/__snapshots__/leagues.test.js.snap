// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues should match the snopshot 1`] = `
<main
  className="main"
>
  <section
    className="league-tabs-container"
  >
    <Connect(LeaguesTabs)
      selectLeaguesData={[Function]}
    />
    <input
      className="search-bar"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </section>
  <section
    className="match-container"
    key="157015"
  >
    <div
      className="match-second-container"
    >
      <p
        className="team-name home-team-name"
      >
        Liverpool
      </p>
      <div
        className="logos-result-container"
      >
        <div
          className="home-team-logo-container"
        >
          <img
            alt="home team logo"
            className="small-logo"
            src="https://www.api-football.com/public/teams/40.png"
          />
        </div>
        <h4
          className="results"
        >
          <span>
             - 
          </span>
        </h4>
        <div
          className="away-team-logo-container"
        >
          <img
            alt="away team logo"
            className="small-logo"
            src="https://www.api-football.com/public/teams/71.png"
          />
        </div>
      </div>
      <p
        className="team-name away-team-name"
      >
        Norwich
      </p>
    </div>
    <p
      className="date"
    >
      2019-08-09
    </p>
    <p
      className="match-status"
    >
      Not Started
    </p>
  </section>
</main>
`;
